### PR TITLE
MySQL support

### PR DESCRIPTION
**Pull request by mruwnik**
This is just the MySQL handling stuff, no actual usage

### CODE REVIEW

File: api/src/stampy_chat/db/session.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+@contextmanager
+def make_session(auto_commit=False):
+    with Session(engine, autoflush=False) as session:
+        yield session
+        if auto_commit:
+            session.commit()
+
+
+class ItemAdder:
+    """A helper class to manage adding and flushing items to the database.
```
Comment (mruwnik) [inline api/src/stampy_chat/db/session.py:25]: this isn't needed here, but I couldn't be bothered to mess about with deleting it and inserting it elsewhere

File: api/src/stampy_chat/db/models.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    query: Mapped[str] = mapped_column(String(1028))
+
+    # The full prompt as sent to the LLM
+    prompt: Mapped[Optional[str]] = mapped_column(LONGTEXT)
+
+    # Whatever the LLM returns
+    response: Mapped[Optional[str]] = mapped_column(LONGTEXT)
+
+    # The ids of the chunks used for the prompt
+    chunks: Mapped[Optional[str]] = mapped_column(String(1028))  # TODO: Change this to a proper format
```
Comment (mruwnik) [inline api/src/stampy_chat/db/models.py:63]: the ids are like 10 chars each, and there shouldn't be that many of them, so 1028 bytes should suffice?

File: local_db.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (mruwnik) [inline local_db.sh:1]: this is exactly the same as in the ARD repo, other than it using a different database. The nice thing about this is that there will only be one MySQL Docker container hanging around, so it should be easy to check things out from the same mysql session

File: api/migrations/alembic.ini [hunk 4]
```diff
@@ -0,0 +1,110 @@
+# A generic, single database configuration.
```
Comment (mruwnik) [inline api/migrations/alembic.ini:1]: boilerplate stuff

**Comment by mruwnik** [comment]
ping

**Comment by mruwnik** [comment]
ping

**Review by ccstan99 — APPROVED** [review]

**Pull request merged**